ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/infrastructure/Assertions.scala

```diff
@@ -189,6 +201,74 @@ object Assertions {
     )
   }
 
+  def assertCompletionIsDefined(optCompletion: Option[Completion]): Completion = {
+    assert(optCompletion.isDefined, "No completion has been produced")
+    optCompletion.get
+  }
+
+  def assertDeduplicationDuration(
+      requestedDeduplicationDuration: Duration,
+      previousSubmissionSendTime: Instant,
+      completionReceiveTime: Instant,
+      completion: Completion,
+      durationSupport: DurationSupport,
+  ): Unit = discard {
```

Comment:
`assert` returns `Unit`. This `discard` call (and the others below) tell us that we are not handling certain cases, as @nicu-da points out. I think we should remove it and fix the errors.